pypeit/core/coadd.py

```diff
@@ -741,22 +757,39 @@ def sn_weights(waves, fluxes, ivars, masks, sn_smooth_npix, const_weights=False,
 
     # Calculate S/N
     sn_val = flux_stack*np.sqrt(ivar_stack)
-    sn_val_ma = np.ma.array(sn_val, mask = np.invert(mask_stack))
+    sn_val_ma = np.ma.array(sn_val, mask=np.logical_not(mask_stack))
     sn_sigclip = stats.sigma_clip(sn_val_ma, sigma=3, maxiters=5)
-    ## TODO Update with sigma_clipped stats with our new cenfunc and std_func = mad_std
-    sn2 = (sn_sigclip.mean(axis=0).compressed())**2 #S/N^2 value for each spectrum
-    rms_sn = np.sqrt(sn2) # Root Mean S/N**2 value for all spectra
+    # TODO: Update with sigma_clipped stats with our new cenfunc and std_func = mad_std
+    sn2 = (sn_sigclip.mean(axis=0).compressed())**2  #S/N^2 value for each spectrum
+    rms_sn = np.sqrt(sn2)  # Root Mean S/N**2 value for all spectra
     spec_vec = np.arange(nspec)
 
-    # TODO: ivar weights is better than SN**2 or const_weights for merging orders. Enventially, we will change it to
-    # TODO Should ivar weights be deprecated??
+    # Check if relative weights input
+    if ivar_weights:
+        relative_weights = False
+    elif relative_weights:
+        if ref_spec == -1:
+            # Relative weights are requested, but no reference spectrum provided.
+            # Use the highest S/N spectrum as a reference
+            ref_spec = np.argmax(sn2)
+            if verbose:
+                msgs.info(
+                    "The reference spectrum (ref_spec={0:d}) has a typical S/N = {1:.3f}".format(ref_spec, sn2[ref_spec]))
+
+        elif (ref_spec < 0) or (ref_spec >= nstack):
+            msgs.error("Provided reference spectrum (ref_spec={0:d}) is not within range 0 <= ref_spec < {1:d}".format(ref_spec, nstack))
+    elif (ref_spec >= 0) and (ref_spec < nstack):
+        if verbose: msgs.info("Computing relative weights")
+        relative_weights = True
```

Comment:
I think you should require relative weights to just be true if the user wants relative weights. That simplifies the logic here, and you can get rid of this last elif.